packages/app/src/systems/Settings/hooks/useSettings.ts

```diff
@@ -0,0 +1,65 @@
+import { useInterpret, useSelector } from '@xstate/react';
+import { useNavigate } from 'react-router-dom';
+
+import type { SettingsMachineState } from '../machines/settingsMachine';
+import { settingsMachine } from '../machines/settingsMachine';
+
+import type { AccountInputs } from '~/systems/Account';
+import { useAccount } from '~/systems/Account';
+import { Pages } from '~/systems/Core';
+
+const selectors = {
+  isChangingPassword: (state: SettingsMachineState) =>
+    state.matches('changingPassword'),
+
+  hasChangedPassword: (state: SettingsMachineState) => state.matches('done'),
+
+  isUnlocking: (state: SettingsMachineState) =>
+    state.matches('unlocking') || state.matches('gettingMnemonic'),
```

Comment:
To avoid matches against states names you can use `tags` inside the machine states